### PR TITLE
dbg-macro fixes for conan2

### PR DESCRIPTION
**Pull request by jabbas**
Specify library name and version:  **dbg-macro/0.4.0**

bumps recipe to conan 2.x format

---

- [*] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [*] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [*] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/dbg-macro/all/conanfile.py [hunk 1]
```diff
@@ -13,31 +15,25 @@ class DbgMacroConan(ConanFile):
...
-        if self.settings.compiler == "gcc" and tools.Version(self.settings.compiler.version) < "5":
+    def configure(self):
+        if self.settings.compiler == "gcc" and int(f"{self.settings.compiler.version}") < 5:
             raise ConanInvalidConfiguration(
-                "dbg-mcro can't be used by {0} {1}".format(
+                "dbg-macro can't be used by {0} {1}".format(
                     self.settings.compiler,
                     self.settings.compiler.version
                 )
             )
```
Comment (RubenRBS) [inline recipes/dbg-macro/all/conanfile.py:32]: This check should also go into the `validate()` method :)

File: recipes/dbg-macro/all/conanfile.py [hunk 2]
```diff
@@ -13,31 +15,25 @@ class DbgMacroConan(ConanFile):
     settings = ("compiler", )
     no_copy_source = True
```
Comment (RubenRBS) [inline recipes/dbg-macro/all/conanfile.py:16]:
```suggestion
    no_copy_source = True
    package_type = "header-library"
```

Also, not a blocker at all, but if you end up having to revisit the changes, please try to remember and remove the `conan` topic entry :)

File: recipes/dbg-macro/all/conanfile.py [hunk 3]
```diff
@@ -13,31 +15,25 @@ class DbgMacroConan(ConanFile):
...
-        extracted_dir = self.name + "-" + self.version
-        os.rename(extracted_dir, self._source_subfolder)
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
 
-    def configure(self):
-        minimal_cpp_standard = "11"
-        if self.settings.get_safe("compiler.cppstd"):
-            tools.check_min_cppstd(self, minimal_cpp_standard)
+    def validate(self):
+        check_min_cppstd(self, 17)
```
Comment (RubenRBS) [inline recipes/dbg-macro/all/conanfile.py:23]: We need to have an extra check this, please check https://github.com/conan-io/conan-center-index/blob/master/docs/package_templates/header_only/all/conanfile.py#L67, it should be straightforward to copy

**Review by RubenRBS — COMMENTED** [review]
Thanks a lot! I have some suggestions and requests, but the changes themselves look good :)

File: recipes/dbg-macro/all/conanfile.py (hunk 1 above)
Comment (jabbas) [reply recipes/dbg-macro/all/conanfile.py]: Thanks @RubenRBS for tip. Done.

File: recipes/dbg-macro/all/conanfile.py (hunk 2 above)
Comment (jabbas) [reply recipes/dbg-macro/all/conanfile.py:16]: Done, wasn't checking that, just left it as is.

Fixed. Thanks.

File: recipes/dbg-macro/all/conanfile.py (hunk 3 above)
Comment (jabbas) [reply recipes/dbg-macro/all/conanfile.py]: Didn't copy and paste it, but made it similarly.

I can't see the point of creating the class property for such simple thing.
I hope you're ok with my attempt.

File: recipes/dbg-macro/all/conanfile.py [hunk 4]
```diff
@@ -9,35 +12,29 @@ class DbgMacroConan(ConanFile):
     homepage = "https://github.com/sharkdp/dbg-macro"
     license = "MIT"
     description = "A dbg(...) macro for C++"
-    topics = ("conan", "debugging", "macro", "pretty-printing", "header-only")
+    topics = ("debugging", "macro", "pretty-printing", "header-only")
     settings = ("compiler", )
```
Comment (RubenRBS) [inline recipes/dbg-macro/all/conanfile.py:16]:
```suggestion
    package_type = "header-library"
    settings = "os", "arch", "compiler", "build_type"
```

File: recipes/dbg-macro/all/test_package/conanfile.py [hunk 5]
```diff
@@ -1,18 +1,27 @@
...
+from conan import ConanFile
+from conan.tools.cmake import CMake, cmake_layout
+from conan.tools.build import cross_building, can_run
+from conan.tools.files import mkdir
 
 
 class TestPackageConan(ConanFile):
     settings = "os", "compiler", "build_type", "arch"
-    generators = "cmake"
+    generators = "CMakeDeps", "CMakeToolchain"
```
Comment (RubenRBS) [inline recipes/dbg-macro/all/test_package/conanfile.py:10]:
```suggestion
    generators = "CMakeDeps", "CMakeToolchain"
    test_type = "explicit"
```

**Review by RubenRBS — COMMENTED** [review]
Thanks a lot! Some more quick changes, but otherwise looks good :)

**Review by RubenRBS — APPROVED** [review]
Thanks a lot! LGTM

**Review by jwillikers — APPROVED** [review]

**Pull request merged**